servatrice/src/serversocketinterface.cpp

```diff
@@ -971,6 +972,39 @@ Response::ResponseCode AbstractServerSocketInterface::cmdBanFromServer(const Com
     return Response::RespOk;
 }
 
+QString AbstractServerSocketInterface::parseEmailAddress(const std::string &stdEmailAddress)
+{
+    QString emailAddress = QString::fromStdString(stdEmailAddress);
+
+    const QRegularExpression rx(R"(\b([A-Za-z0-9._%+-]+)@([A-Za-z0-9.-]+\.[A-Za-z]{2,4})\b)");
+    const QRegularExpressionMatch match = rx.match(emailAddress);
+
+    if (emailAddress.isEmpty() || !match.hasMatch()) {
+        return QString();
+    }
+
+    const QString capturedEmailAddressDomain = match.captured(2);
+
+    // Trim out dots and pluses from Google/Gmail domains
+    if (capturedEmailAddressDomain.contains("gmail.com", Qt::CaseInsensitive) ||
```

Comment:
why not use a simple comparison? this matches frogmail.com or something like it

Reply:
Fixed